consensus/core/src/tests/randomized_tests.rs

```diff
@@ -0,0 +1,233 @@
+    let wave_number =
+        authority_1.committer.committers[pipeline].wave_number(random_test_setup.num_waves as u32);
+    let num_rounds = authority_1.committer.committers[pipeline].decision_round(wave_number);
+    let include_leader_percentage = 50;
+    let dag_builder = create_random_dag(
+        random_test_setup.seed,
+        include_leader_percentage,
+        num_rounds,
+        authority_1.context.clone(),
+    );
+
+    tracing::info!(
+        "Running test with committee size {} & {} completed waves in the DAG...",
+        random_test_setup.num_authorities,
+        random_test_setup.num_waves
+    );
+
+    let mut all_blocks = dag_builder.blocks.values().cloned().collect::<Vec<_>>();
+    all_blocks.shuffle(&mut random_test_setup.seeded_rng);
+
+    let mut sequenced_leaders_1 = vec![];
+    let mut last_decided = Slot::new_for_test(0, 0);
+    for block in &all_blocks {
+        let _ = authority_1
+            .block_manager
+            .try_accept_blocks(vec![block.clone()]);
+        let sequence = authority_1.committer.try_decide(last_decided);
+
+        if !sequence.is_empty() {
+            sequenced_leaders_1.extend(sequence.clone());
+            let leader_status = sequence.last().unwrap();
+            last_decided = Slot::new(leader_status.round(), leader_status.authority());
+        }
+    }
+
+    assert!(authority_1.block_manager.is_empty());
+
+    // Setup for Authority 2
+    let mut authority_2 = authority_setup(random_test_setup.num_authorities);
+
+    let mut all_blocks = dag_builder.blocks.values().cloned().collect::<Vec<_>>();
+    all_blocks.shuffle(&mut random_test_setup.seeded_rng);
+
+    let mut sequenced_leaders_2 = vec![];
+    let mut last_decided = Slot::new_for_test(0, 0);
+    for block in &all_blocks {
+        let _ = authority_2
+            .block_manager
+            .try_accept_blocks(vec![block.clone()]);
+        let sequence = authority_2.committer.try_decide(last_decided);
+
+        if !sequence.is_empty() {
+            sequenced_leaders_2.extend(sequence.clone());
+            let leader_status = sequence.last().unwrap();
+            last_decided = Slot::new(leader_status.round(), leader_status.authority());
+        }
+    }
+
+    assert!(authority_2.block_manager.is_empty());
+
+    // Ensure despite the difference in when blocks were received eventually after
+    // receiving all blocks both authorities should return the same sequence of blocks.
+    assert_eq!(sequenced_leaders_1, sequenced_leaders_2);
+}
+
+struct AuthorityTestFixture {
+    context: Arc<Context>,
+    dag_state: Arc<RwLock<DagState>>,
+    committer: UniversalCommitter,
+    block_manager: BlockManager,
+}
+
+fn authority_setup(num_authorities: usize) -> AuthorityTestFixture {
+    let context = Arc::new(
+        Context::new_for_test(num_authorities)
+            .0
+            .with_authority_index(AuthorityIndex::new_for_test(1)),
```

Comment:
I don't believe that this plays a role for those kind of tests, but maybe we want to allow pass in the authority index?